pkg/ccl/backupccl/restore.go

```diff
@@ -1579,6 +1579,9 @@ type restoreResumer struct {
 	tables      []*sqlbase.TableDescriptor
 	latestStats []*stats.TableStatisticProto
 	execCfg     *sql.ExecutorConfig
+	// A collection of stages that should only be run once.
+	statsInserted   bool
```

Comment:
don't you need to persist these, in the job detail or progress?

Reply:
yep. looks like this was the wrong branch.

Reply:
👍 
I think you might need to actually update the job too to save your change to the in-mem version you're updating here. You could do it in the same txn ideally I think?

Reply:
Here's how IMPORT does something similar (ensuring that all the descs get their state changed): https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/importccl/import_stmt.go#L900